src/policy/fees.cpp

```diff
@@ -663,6 +663,23 @@ void CBlockPolicyEstimator::processBlock(unsigned int nBlockHeight,
     shortStats->UpdateMovingAverages();
     longStats->UpdateMovingAverages();
 
+    // Do not consider transactions that were CPFP'd.
+    // For each mempool transaction that was included in a block, if its individual feerate is higher
+    // than its ancestor score it incentivized the inclusion of some of its parents in the block.
+    // Remove all such parents whose feerate is lower than the child's of the tracking before updating
+    // the data points.
+    for (const auto& entry : entries) {
+        CFeeRate individual_feerate{entry->GetFee(), static_cast<uint32_t>(entry->GetTxSize())};
+        CFeeRate ancestor_score{entry->GetModFeesWithAncestors(), static_cast<uint32_t>(entry->GetSizeWithAncestors())};
+
+        if (individual_feerate > ancestor_score) {
+            for (const CTxMemPoolEntry& parent : entry->GetMemPoolParents()) {
+                CFeeRate parent_feerate{parent.GetFee(), static_cast<uint32_t>(parent.GetTxSize())};
+                if (parent_feerate < individual_feerate) _removeTx(parent.GetTx().GetHash(), /* inBlock = */true);
```

Comment:
If we had a chain:

```
A->B-C
2   1  100 sat/vbyte
```
Would this mean that `A` would be considered as not a CPFP by this logic?

Reply:
Yes. This code would only catch the most simple one-parent one-child CPFPs. Do you think we should do better? I'm a bit wary of starting to consider the child as then we'd have to consider its own ancestor feerate to avoid false positives.

Reply:
Another option is to consider 3+ chains as usual, without dropping the child. (CPFP implementations use RBF, right?)
Although I guess it won't be worse than today anyway, assuming someone really wants to trigger poor estimation on purpose?

I agree this should be better documented.